### ISSUE TITLE
 LoadError: UndefVarError: metadatasupport not defined

### ISSUE THREAD

**Issue by dalejordan** [JuliaData/DataFrames.jl#3243]
It looks like metadatasupport and colmetadatasupport are not imported by DataFrames.jl, resulting in the above precompilation error.
v1.4.4

**Comment by bkamins** [comment]
Are sure your DataAPI.jl version is 1.13.0? (maybe you have not properly instantiated your package environemnt?)
On my machine DataFrames.jl 1.4.4 works fine.

**Comment by bkamins** [comment]
Also note that DataFrames.jl does not import not export `metadatasupport` and `colmetadatasupport`, it just defines methods for these functions here:
https://github.com/JuliaData/DataFrames.jl/blob/330f9d3e458fb076902f3e4ceda3367f15ec4f58/src/other/metadata.jl#L7-L15

**Comment by dalejordan** [comment]
Yes, my DataAPI is 1.13.0.  I was referring to the fact that the main file
DataFrames.jl imports a bunch of functions
from there, but does not import metadatasupport, leading to an UndefVar
error when precompiling DataFrames.
The needed functions are defined in DataAPI.

**Comment by bkamins** [comment]
Can you please show me the exact stack trace of the error? We never call unqualified `metadtasupport`. Everywhere this function is called we use `DataAPI` prefix so importing it should not be needed.

The 1.4.4 version has around 10000 installations since the release and we have not had such a report, so your configuration must be non standard and I would like to investigate the reason. Thank you!

**Comment by dalejordan** [comment]
Here's the stacktrace:
```
(example) julia> include("GTMs/example/iris.jl");
[ Info: Precompiling DataFrames [a93c6f00-e57d-5684-b7b6-d8193f3e46c0]
ERROR: LoadError: UndefVarError: metadatasupport not defined
Stacktrace:
 [1] getproperty(x::Module, f::Symbol)
   @ Base ./Base.jl:31
 [2] top-level scope
   @ ~/.julia/packages/DataFrames/dgZn3/src/other/metadata.jl:8
 [3] include(mod::Module, _path::String)
   @ Base ./Base.jl:419
 [4] include(x::String)
   @ DataFrames ~/.julia/packages/DataFrames/dgZn3/src/DataFrames.jl:1
 [5] top-level scope
   @ ~/.julia/packages/DataFrames/dgZn3/src/DataFrames.jl:172
 [6] include
   @ ./Base.jl:419 [inlined]
 [7] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl
_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, 
UInt64}}, source::String)
   @ Base ./loading.jl:1554
 [8] top-level scope
   @ stdin:1
in expression starting at /home/dale/.julia/packages/DataFrames/dgZn3/src/other/metadata.jl:8
in expression starting at /home/dale/.julia/packages/DataFrames/dgZn3/src/DataFrames.jl:1
in expression starting at stdin:1
ERROR: LoadError: Failed to precompile DataFrames [a93c6f00-e57d-5684-b7b6-d8193f3e46c0] to /h
ome/dale/.julia/compiled/v1.8/DataFrames/jl_0DxNrh.
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:35
  [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::IO, internal_stdout::IO, ke
ep_loaded_modules::Bool)
    @ Base ./loading.jl:1707
  [3] compilecache
    @ ./loading.jl:1651 [inlined]
  [4] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:1337
  [5] _require_prelocked(uuidkey::Base.PkgId)
    @ Base ./loading.jl:1200
  [6] macro expansion
    @ ./loading.jl:1180 [inlined]
  [7] macro expansion
    @ ./lock.jl:223 [inlined]
  [8] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:1144
  [9] include(fname::String)
    @ Base.MainInclude ./client.jl:476
 [10] top-level scope
    @ REPL[2]:1
in expression starting at /home/dale/jl/GTMs/example/iris.jl:4
```
My comments on the lack of import of metadatasupport were based of the
source code using the syntax import 'DataAPI: symbol,...' .  It was my understanding
that once you do that it is no longer sufficient to use 'DataAPI.symbol' to define a method, it must be in the import list.
I might be wrong...

**Comment by bkamins** [comment]
Your error message indicates that something is wrong with DataAPI.jl version. Are you sure you are on 1.13.0?
Here is an example that shows that what is defined is correct (run in a clean environment):
```
(@v1.8) pkg> st DataAPI
Status `~\.julia\environments\v1.8\Project.toml`
  [9a962f9c] DataAPI v1.13.0

julia> import DataAPI, DataAPI.metadata # I additionally import DataAPI.metadata to show that extra imports do not matter here

julia> DataAPI.metadatasupport(::Type{<:Vector}) = "Hello"

julia> DataAPI.metadatasupport(typeof([1, 2, 3]))
"Hello"

julia> methods(DataAPI.metadatasupport)
# 2 methods for generic function "metadatasupport":
[1] metadatasupport(::Type{<:Vector}) in Main at REPL[3]:1
[2] metadatasupport(::Type) in DataAPI at ~\.julia\packages\DataAPI\ILCXt\src\DataAPI.jl:316
```
(as you can see such imports work)

So can you please recheck that you are on DataAPI 1.13.0 (by e.g. running the code I used above and showing the output?)

Thank you!

**Comment by dalejordan** [comment]
OK, I think I have uncovered the problem and (inadvertantly) fixed it.
 After updating my project environment to download 1.4.4 I restarted Julia and it immediately 
precompiled OhMyREPL, PackageTemplates, and Revise.
I then used DataFrames (in my project env.) and it errored out precompiling.
Attempting to follow your example in your previous post, I added DataAPI to my @v1.8 env,
which promptly upgraded it from v1.11.0 to V1.13.0 and then (for whatever reason) precompiled
DataFrames (successfully).
I think one of the three packages above must have loaded the v1.11.0 DataAPI which prevented
DataFrames from loading v1.13.0.

I misread the import statement in DataFrames.jl, so my original comments were wrong; sorry
for the noise on that score.  I thank you for your attention and persistence to get to the bottom
of the problem.
